src/lockscreen.js

```diff
@@ -1,51 +1,60 @@
 /* exported Lockscreen */
 
-const {Gdk, Gio, GLib, GObject, Gtk} = imports.gi;
+const {Gio, GObject, Gtk} = imports.gi;
 
-const FADE_OUT_TIME_MS = 750;
+const {Playbin} = imports.playbin;
 
 var Lockscreen = GObject.registerClass({
     GTypeName: 'Lockscreen',
+    CssName: 'lockscreen',
     Properties: {
         locked: GObject.ParamSpec.boolean('locked', 'Locked', '',
             GObject.ParamFlags.READWRITE | GObject.ParamFlags.CONSTRUCT,
             true),
         key: GObject.ParamSpec.string('key', 'Key', '',
-            GObject.ParamFlags.READWRITE, ''),
+            GObject.ParamFlags.READWRITE, null),
         lock: GObject.ParamSpec.string('lock', 'lock', '',
-            GObject.ParamFlags.READWRITE, ''),
-    },
-    Signals: {
-        'overlay-clicked': {},
+            GObject.ParamFlags.READWRITE, null),
     },
 }, class Lockscreen extends Gtk.Overlay {
     _init(props = {}) {
+        this._locked = true;
+        this._key = null;
+        this._lock = null;
+
         super._init(props);
 
-        this._overlay = new Gtk.EventBox({
+        this._playbin = new Playbin({
             expand: true,
-            visible: true,
+            noShowAll: true,
+            visible: false,
+        });
+
+        this._playbin.connect('clicked', this._onClicked.bind(this));
+
+        this._playbin.connect('start', () => {
+            this._playbin.remove_class('locked');
         });
-        this._overlay.get_style_context().add_class('lockscreen');
-        this._updateUI();
 
-        this._overlay.connect('button-release-event', () => {
-            this.emit('overlay-clicked');
-            return Gdk.EVENT_PROPAGATE;
+        this._playbin.connect('done', () => {
+            this._playbin.stop();
+            this._playbin.hide();
         });
 
+        this.add_overlay(this._playbin);
+
         this._manager = Gio.Application.get_default().locksManager;
         this._keyChangedId = 0;
         this._lockChangedId = 0;
-        this.connect('overlay-clicked', this._onClicked.bind(this));
+        this._updateUI();
     }
 
     get locked() {
         return this._locked;
     }
 
     set locked(value) {
-        if ('_locked' in this && this._locked === value)
+        if (this._locked === value)
```

Comment:
I'd like to preserve the original `if('$prop' in this)` pattern here, unless there's a reason not to